test/sharness/t0040-add-and-cat.sh

```diff
@@ -78,6 +78,56 @@ test_add_cat_file() {
     test_expect_code 1 ipfs cat --offset -102 "$HASH" > actual
   '
 
+  test_expect_success "ipfs cat with length succeeds" '
+    ipfs cat --length 8 "$HASH" >actual
+  '
+
+  test_expect_success "ipfs cat with length output looks good" '
+    echo -n "Hello Wo" >expected &&
```

Comment:
Note the `-n` flag is not universally supported by all shells (for example on the mac).  It is better to use `printf` which is more portable.  In this case it will be
```
  printf "Hello Wo" >expected &&
```